tests/unit/mainwindow/test_tabwidget.py

```diff
@@ -52,3 +52,18 @@ def test_small_icon_doesnt_crash(self, widget, qtbot, fake_web_tab):
 
         with qtbot.waitExposed(widget):
             widget.show()
+
+    def test_update_tab_titles_benchmark(self, benchmark, widget,
+                                         qtbot, fake_web_tab):
+        """Benchmark for update_tab_titles."""
+        widget.addTab(fake_web_tab(), 'foobar')
+        widget.addTab(fake_web_tab(), 'foobar2')
+        widget.addTab(fake_web_tab(), 'foobar3')
+        widget.addTab(fake_web_tab(), 'foobar4')
+
+        with qtbot.waitExposed(widget):
+            widget.show()
+
+        def bench():
+            widget._update_tab_titles()
+        benchmark(bench)
```

Comment:
No need to define an extra function now - just do `benchmark(widget._update_tab_titles)` :wink:

Also, I guess without the `for` loop, the benchmark outcome still looks like you'd expect?

Reply:
oh, I keep forgetting I can do that! :)

Yup, the benchmark takes about 800ms, which seems like it's repeating it properly!